test/test_settings.py

```diff
@@ -342,11 +342,22 @@ def test_cannot_set_invalid_values_for_max_header_list_size(self, val):
             assert e.value.error_code == h2.errors.ErrorCodes.PROTOCOL_ERROR
             assert s.max_header_list_size is None
 
-            with pytest.raises(h2.exceptions.InvalidSettingsValueError) as e:
+            with pytest.raises(h2.exceptions.InvalidSettingsValueError) as e: 
                 s[h2.settings.MAX_HEADER_LIST_SIZE] = val
 
             s.acknowledge()
             assert e.value.error_code == h2.errors.ErrorCodes.PROTOCOL_ERROR
 
             with pytest.raises(KeyError):
                 s[h2.settings.MAX_HEADER_LIST_SIZE]
+
+
+    def equal_objects_are_equal(s1,s2):
+        s1.acknowledge()
+        s2.acknowledge()
+        assert s1._settings==s2._settings
+   
+    def notequal_objects_are_notequal(s1,s2):
+        s1.acknowledge()
+        s2.acknowledge()
+        assert s1._settings!=s2._settings
```

Comment:
Ok, so these tests cannot possibly work right now for a number of reasons. Let's discuss why.

Firstly, and most importantly, where are `s1` and `s2` coming from? Nowhere else in the code knows how to provide them. If you look at tests higher up the file you'll see they create their own `Settings` objects: your tests need to do that as well.

Secondly, you need to think a bit harder about what these tests are actually testing. What you want to do is to confirm that the code you just wrote *actually works*. That means you want to test that two Settings entities that are the same are equal, and two that are different are not equal.

As an example, the simplest test that meaningfully tests something looks like this:

```python
def test_newly_created_settings_are_equal(self):
    s1 = h2.settings.Settings()
    s2 = h2.settings.Settings()

    assert s1 == s2
```

Note as well some basic rules of Python code: specifically, note that these test methods are *methods*, and so must take a `self` parameter, and note as well that to have them count as a test they need to have names that begin with`test_`.